### PR TITLE
Python ML Process connected to backend movies

### PR DESCRIPTION
**Pull request by Capn05**
Created JS script "FetchMLRecs" which when ran fetches user movies and passes it as an argument to the python ML Process.

Changes to ML process "svd_movie_ratings" include:
appending new user data to static psuedo user data utilizing pandas dataframe
Running SVD predictions on new pandas dataframe
returning top 'n' predictions for a specific user (hard coded currently)

ML process takes 1 minute to run so far
Screenshot 2024-07-17 at 10 29 48 AM

Next Step:
Connect returned predictions to front end to showcase to user

### CODE REVIEW

File: scripts/FetchMLRecs.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        });
+        return ratings.map(r => ({
+            userID: r.userId,
+            itemID: r.movie.movieId,
+            rating: r.rating/2,
+            timestamp: r.timestamp
+        }));
+
+}
+import { spawn }from 'child_process';
```
Comment (samyalihamad) [inline scripts/FetchMLRecs.js:25]: Typically all imports are defined at the top of the page

File: scripts/FetchMLRecs.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                        movieId: true
+                    }
+                },
+                rating: true,
+            }
+        });
+        return ratings.map(r => ({
+            userID: r.userId,
+            itemID: r.movie.movieId,
+            rating: r.rating/2,
```
Comment (samyalihamad) [inline scripts/FetchMLRecs.js:20]: Why divide by 2?

File: svd_movie_ratings.py [hunk 3]
```diff
@@ -2,64 +2,86 @@
...
+
+        Returns:
+        A dict where keys are user (raw) ids and values are lists of tuples:
+            [(raw item id, rating estimation), ...] of size n.
+        """
+
+        # First map the predictions to each user.
+        top_n = defaultdict(list)
+        for uid, iid, true_r, est, _ in predictions:
+            if (uid == 'rOoKKGzDFxa2Av7KogZVszUTJuI3'):
```
Comment (samyalihamad) [inline svd_movie_ratings.py:38]: What's this comparison doing?

File: scripts/FetchMLRecs.js (hunk 1 above)
Comment (Capn05) [reply scripts/FetchMLRecs.js:25]: Good point, fixed!

File: scripts/FetchMLRecs.js (hunk 2 above)
Comment (Capn05) [reply scripts/FetchMLRecs.js:20]: The ratings in the database are out of 5 but mine are out of 10

File: svd_movie_ratings.py (hunk 3 above)
Comment (Capn05) [reply svd_movie_ratings.py:38]: This was a hardcoded user ID for testing recommendations, It has not been changed to be dynamic

**Pull request merged**